src/sisl/io/_multiple.py

```diff
@@ -0,0 +1,236 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+from functools import (
+        update_wrapper,
+        reduce
+)
+from textwrap import dedent
+from typing import Any, Callable, Optional, Type
+from numbers import Integral
+
+Func = Callable[..., Optional[Any]]
+
+
+class SileSlicer:
+    """ Handling io-methods in sliced behaviour for multiple returns
+
+    This class handler can expose a slicing behavior of the function
+    that it applies to.
+
+    The idea is to attach a function/method to this class and
+    let this perform the function at hand for slicing behaviour
+    etc.
+    """
+    def __init__(self,
+                 obj: Type[Any],
+                 func: Func,
+                 key: Type[Any],
+                 *,
+                 skip_func: Optional[Func]=None,
+                 postprocess: Optional[Callable[..., Any]]=None):
+        # this makes it work like a function bound to an instance (func.__self__
+        # works for instances)
+        self.__self__ = obj
+        self.__func__ = func
+        self.key = key
+        if skip_func is None:
+            self.skip_func = func
+        else:
+            self.skip_func = skip_func
+        self.postprocess = postprocess
```

Comment:
What happens if `postprocess is None`? Could `list` be a default?

Reply:
It then returns as the decorated routine returns the values (in this case a list of the return values).
I see there was no default, now it is `lambda x: x` (or the equivalent).

So yes, the default is a list.